parser_library/src/workspaces/parse_lib_provider.h

```diff
@@ -39,9 +42,8 @@ class parse_lib_provider
 
     virtual bool has_library(const std::string& library, const utils::resource::resource_location& program) const = 0;
 
-    virtual std::optional<std::string> get_library(const std::string& library,
-        const utils::resource::resource_location& program,
-        std::optional<utils::resource::resource_location>& lib_location) const = 0;
+    virtual std::optional<std::pair<std::string_view, utils::resource::resource_location>> get_library(
```

Comment:
I don't mind the switch from out parameter to rerturning a pair, however the change from string to string_view means that somebody else is owning the actual value.
That's not the direction I would like it to go - it makes the life time reasoning more complicated and opaque.